pkg/process/checks/net.go

```diff
@@ -149,11 +151,22 @@ func (c *ConnectionsCheck) getConnections() (*model.Connections, error) {
 	return tu.GetConnections(c.tracerClientID)
 }
 
-func (c *ConnectionsCheck) getLastConnectionsByPID() map[int32][]*model.Connection {
-	if result := c.lastConnsByPID.Load(); result != nil {
-		return result.(map[int32][]*model.Connection)
+func (c *ConnectionsCheck) notifyProcessConnRates(conns *model.Connections) {
+	if len(c.processConnRatesTransmitter.Chs) == 0 {
+		return
 	}
-	return nil
+
+	connCheckIntervalS := int(GetInterval(ConnectionsCheckName) / time.Second)
+
+	connByPID := getConnectionsByPID(conns)
```

Comment:
Can we do this in one pass? I think we can go over the connections once and populate `connRates` as we go? Would have to just keep a count of the connections per pid.

Reply:
Thank you @hmahmood, to confirm this is what you meant - we'd need to change the structs we return from here then, since we wouldn't be able to consume into `model.ProcessNetworks`:
```
message ProcessNetworks {
	float connectionRate = 1;
	float bytesRate = 2;
}
```

 it is because of this:
```
connRate := float32(len(conns)) / float32(interval)
bytesRate := float32(totalTraffic) / float32(interval)
```
So we wouldn't know when the collection for a given PID is complete.

With that in mind the struct we would end up with is:
```
type ProcessConnData struct {
  TotalConnections	int64
  TotalBytes        int64
  Interval          time.Duration
}

Can we hold these changes until we start consuming them in the Process check - right now this code is going to continue to be dormant:
```
if len(c.processConnRatesTransmitter.Chs) == 0 {
		return
}
```

(+ backend does not use these fields yet, we may want to separate `bytesRate` into separate fields when we do).

Reply:
Something like this:

Reply:
Aha, ok, so it's the allocation of connections we'd avoid, not the second loop, thank you 🙇

Reply:
Updated, thank you so much for the suggestion @hmahmood!